src/c_api/init.cpp

```diff
@@ -40,6 +41,7 @@ void dlaf_initialize(int argc_pika, const char** argv_pika, int argc_dlaf,
 void dlaf_finalize() noexcept {
   if (dlaf_initialized) {
     pika::resume();
+    dlaf_free_all_grids();
```

Comment:
I think this is ok for now, but we might want to consider putting this inside `dlaf::finalize()` instead? At the moment CP2K uses DLA-Future purely through the fortran interface, but in case there would be future mixed C++/C/Fortran usage, and the cleanup is done only through `dlaf::finalize`, it'd be good to already have it in `dlaf::finalize`. I don't have an immediate use case for this, but it feels like the right thing to do. What do you think? Does it complicate other things to do it like that?

Reply:
As discussed in person, I removed this and the function for free all grids will have to be called explicitly.